ote/src/cljs/ote/views/admin/interfaces.cljs

```diff
@@ -117,10 +118,84 @@
+                                                                                :on-click #(do (.preventDefault %)
+                                                                                               (e! (admin-controller/->OpenOperatorModal interface)))}
+                                                                            operator-name]]
+   [ui/table-row-column {:style {:width "14%" :padding "0px 5px 0px 5px"}} (admin-controller/format-interface-content-values data-content)]
+   [ui/table-row-column {:style {:width "8%" :padding "0px 5px 0px 5px"}} (first format)]
+   [ui/table-row-column {:style {:width "35%" :padding "0px 5px 0px 5px"}} (if (= "Rajapinta puuttuu" import-error)
+                                                                             url
+                                                                             [linkify url url {:target "_blank"}])]
+   [ui/table-row-column {:style {:width "14%" :padding "0px 5px 0px 5px"}}
+    (if (= date-0 imported)
+      ""
+      (time/format-timestamp-for-ui imported))]
+   [ui/table-row-column {:style {:width "10%" :padding "0px 5px 0px 5px"}}
+    (when (and import-error (not= "Rajapinta puuttuu" import-error))
+      [:a {:style {:color "red"}
+           :href "#"
+           :on-click #(do (.preventDefault %)
+                          (e! (admin-controller/->OpenInterfaceErrorModal interface)))}
+       " Latausvirhe "])
+    (when (and db-error (not= "no-db" db-error))
+      [:a {:style {:color "red"}
+           :href "#"
+           :on-click #(do (.preventDefault %)
+                          (e! (admin-controller/->OpenInterfaceErrorModal interface)))}
+       " Käsittelyvirhe "])
+
+    (when (and import-error (= "Rajapinta puuttuu" import-error))
+      [:span " Rajapinta puuttuu kokonaan "])
+
+    (when (and (nil? import-error) (nil? db-error)) [gtfs-viewer-link url (first format)])]])
+
 (defn interface-list [e! app]
   (let [{:keys [loading? results filters]}
         (get-in app [:admin :interface-list])
-        date-0 (js/Date. 0)]
+        date-0 (js/Date. 0)
+        grouped-results (group-by :interface-id results)
+        ;; Take first from every vector
+        first-from-grouped-results (map
+                                     (fn [x]
+                                       (merge {:first? true} x))
+                                     (map #(first (second %)) grouped-results))
+        selected-interface-id (get-in app [:admin :interface-list :selected-interface-id])
+        selected-interfaces (filter
+                             (fn [x]
+                               (= selected-interface-id (:interface-id x)))
+                             results)
+        rows (doall
+                (mapcat (fn [{:keys [interface-id operator-name format data-content url imported import-error db-error created first?]
+                              :as interface}]
+                          [^{:key (str interface created)}
```

Comment:
There's one key already added in `interface-table-row`. 
Maybe that should be kept and this row's key removed, or is it really needed?

Reply:
nice! I'll remove the wrong one